src/main/java/org/threeten/extra/DayTime.java

```diff
@@ -0,0 +1,393 @@
+        return (amountToSubtract == Long.MIN_VALUE ? plus(Long.MAX_VALUE, unit).plus(1, unit) : plus(-amountToSubtract, unit));
+    }
+
+    @Override
+    public long until(final Temporal endExclusive, final TemporalUnit unit) {
+        final DayTime endDayTime = DayTime.from(endExclusive);
+        final long timeUntil;
+
+        if (unit instanceof ChronoUnit) {
+            final long nanosUntil = ((((endDayTime.dayOfWeek.getValue() - dayOfWeek.getValue()) * NANOS_PER_DAY) +
+                    endDayTime.localTime.toNanoOfDay() - localTime.toNanoOfDay()) + NANOS_PER_WEEK) % NANOS_PER_WEEK;
+
+            switch ((ChronoUnit) unit) {
+                case NANOS: {
+                    timeUntil = nanosUntil;
+                    break;
+                }
+
+                case MICROS: {
+                    timeUntil = nanosUntil / 1000;
+                    break;
+                }
+
+                case MILLIS: {
+                    timeUntil = nanosUntil / 1_000_000;
+                    break;
+                }
+
+                case SECONDS: {
+                    timeUntil = nanosUntil / NANOS_PER_SECOND;
+                    break;
+                }
+
+                case MINUTES: {
+                    timeUntil = nanosUntil / NANOS_PER_MINUTE;
+                    break;
+                }
+
+                case HOURS: {
+                    timeUntil = nanosUntil / NANOS_PER_HOUR;
+                    break;
+                }
+
+                case HALF_DAYS: {
+                    timeUntil = nanosUntil / (12 * NANOS_PER_HOUR);
+                    break;
+                }
+
+                case DAYS: {
+                    timeUntil = nanosUntil / NANOS_PER_DAY;
+                    break;
+                }
+
+                default: {
+                    throw new UnsupportedTemporalTypeException("Unsupported unit: " + unit);
+                }
+            }
+        } else {
+            timeUntil = unit.between(this, endDayTime);
+        }
+
+        return timeUntil;
+    }
+
+    @Override
+    public int compareTo(final DayTime other) {
+        final int dayOfWeekComparison = dayOfWeek.compareTo(other.dayOfWeek);
+        return dayOfWeekComparison == 0 ? localTime.compareTo(other.localTime) : dayOfWeekComparison;
+    }
+
+    @Override
+    public boolean equals(final Object other) {
+        if (this == other) {
+            return true;
+        }
+
+        if (other == null || getClass() != other.getClass()) {
+            return false;
+        }
+
+        final DayTime that = (DayTime) other;
+        return dayOfWeek == that.dayOfWeek && Objects.equals(localTime, that.localTime);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(dayOfWeek, localTime);
+    }
+
+    @Override
+    public String toString() {
+        return dayOfWeek.getDisplayName(TextStyle.FULL, Locale.getDefault()) + " " + localTime.toString();
```

Comment:
Now that I'm looking at it, this probably shouldn't be locale-sensitive; if we want locale-sensitivity, that probably belongs in a dedicated "get for display" method that takes a `Locale` as an argument.